shared/node.c

```diff
@@ -502,11 +502,23 @@ static void grok_node(struct cfg_comp *c, merlin_node *node)
 				node->encrypted=false;
 			}
 		} else if (!strcmp(v->key, "publickey")) {
-			if ( open_encryption_key(v->value, node->pubkey,
+			unsigned char pubkey[crypto_box_PUBLICKEYBYTES];
+			if ( open_encryption_key(v->value, pubkey,
 						crypto_box_PUBLICKEYBYTES) ) {
 				cfg_error(c,v, "Could not open publickey: %s\n",
 						v->value);
 			}
+			/* pre-calculate key */
+			if (crypto_box_beforenm(node->sharedkey, pubkey,
+                        ipc.privkey) != 0) {
+				cfg_error(c,v, "Could not pre-calculated shared key\n");
```

Comment:
The message sentence reads weird. Missing a verb before "pre-calculated", or maybe it should be "pre-calculate" ?